src/archive.rs

```diff
@@ -74,43 +82,132 @@ impl ProviderArchive {
+    /// one of the `try_load*_one` methods if you only want to load a single binary into memory.
+    ///
+    /// Please note that this method requires that you have _all_ of the provider archive bytes in
+    /// memory, which will likely be really hefty if you are just trying to load a specific binary
+    /// to run
+    pub async fn try_load(input: &[u8]) -> Result<ProviderArchive> {
+        let mut cursor = Cursor::new(input);
+        Self::load(&mut cursor, None).await
+    }
+
+    /// Attempts to read a Provider Archive (PAR) file's bytes to analyze and verify its contents,
+    /// loading _only_ the specified target.
+    ///
+    /// This is useful when loading a provider archive for consumption and you know the target OS
+    /// you need. The embedded claims in this archive will be validated, and the file hashes
+    /// contained in those claims will be compared and verified against hashes computed at load
+    /// time. This prevents the contents of the archive from being modified without the embedded
+    /// claims being re-signed
+    ///
+    /// Please note that this method requires that you have _all_ of the provider archive bytes in
+    /// memory, which will likely be really hefty if you are just trying to load a specific binary
+    /// to run
+    pub async fn try_load_one(input: &[u8], target: &str) -> Result<ProviderArchive> {
+        let mut cursor = Cursor::new(input);
+        Self::load(&mut cursor, Some(target)).await
+    }
+
+    /// Attempts to read a Provider Archive (PAR) file to analyze and verify its contents.
+    ///
+    /// The embedded claims in this archive will be validated, and the file hashes contained in
+    /// those claims will be compared and verified against hashes computed at load time. This
+    /// prevents the contents of the archive from being modified without the embedded claims being
+    /// re-signed. This will load all binaries into memory in the returned `ProviderArchive`. Use
+    /// one of the `try_load*_one` methods if you only want to load a single binary into memory.
+    pub async fn try_load_file(path: impl AsRef<Path>) -> Result<ProviderArchive> {
+        let mut file = File::open(path).await?;
+        Self::load(&mut file, None).await
+    }
+
+    /// Attempts to read a Provider Archive (PAR) file to analyze and verify its contents.
+    ///
+    /// The embedded claims in this archive will be validated, and the file hashes contained in
+    /// those claims will be compared and verified against hashes computed at load time. This
+    /// prevents the contents of the archive from being modified without the embedded claims being
+    /// re-signed. This will only read a single binary into memory.
+    ///
+    /// It is recommended to use this method or the [`load`] method when consuming a provider
+    /// archive. Otherwise all binaries will be loaded into memory
+    pub async fn try_load_file_one(
```

Comment:
Minor nit: can this be renamed to something a little more self documenting like try_load_target_from_file or something like that?

Reply:
Ok, updated it